src/javascript/app_2/App/Components/Form/digit_selector.jsx

```diff
@@ -0,0 +1,51 @@
+import PropTypes from 'prop-types';
+import React     from 'react';
+
+const DigitSelector = ({
+    name,
+    onChange,
+    selected_digit,
+}) => {
+    const handleSelect = (item) => {
+        if (+item.target.getAttribute('data-value') !== selected_digit) {
+            onChange({ target: { name, value: +item.target.getAttribute('data-value') } });
+        }
+    };
+
+    return (
+        <div className='digit-selector center-text'>
+            <div>
+                {[...Array(5).keys()].map(i =>
```

Comment:
to prevent repetition perhaps we can make a function for this and reuse at line 30? `makeRow(rowItems, rowIndexStart)...` ?

Reply:
why not just store it as a variable `const arr_five = [...Array(5).keys()];`? mapping doesn't change the original value of it and it can be reused